services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/docker_utils.py

```diff
@@ -94,58 +95,98 @@ async def pull_images(
         )
 
 
+#
+# HELPERS
+#
 _DOWNLOAD_RATIO: Final[float] = 0.75
-_LayerInfoDict = dict[str, dict[str, tuple[int, int]]]
-_PULL_PROGRESS_STATES: Final[list[str]] = [
-    "Downloading",
-    "Download complete",
-    "Extracting",
-    "Pull complete",
-]
+
+LayerId = str
+_LayersInfoDict = dict[LayerId, tuple[int, int]]
+ImageName = str
+_ImagesInfoDict = dict[ImageName, _LayersInfoDict]
+
+
+class _ProgressDetailDict(TypedDict, total=False):
+    current: int
+    total: int
+
+
+class _DockerProgressDict(TypedDict, total=False):
```

Comment:
ok just wondering if you would maybe look into packages/models-library/src/models_library/generated_models/docker_rest_api.py, you might find these models already

Reply:
aha, i did not look carefully. I will check again.